vector/src/main/java/im/vector/home/NewHomeFragment.kt

```diff
@@ -0,0 +1,185 @@
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(getLayoutResId(), container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        pager.offscreenPageLimit = 3
+        pager.adapter = HomePagerAdapter(childFragmentManager, resources.getStringArray(R.array.home_tabs))
+        tabLayout.setupWithViewPager(pager)
+        activity?.let { activity ->
+            setTabLayoutTheme(activity, tabLayout)
+            mPrimaryColor = getColor(activity, R.attr.vctr_tab_home)
+            mSecondaryColor = getColor(activity, R.attr.vctr_tab_home_secondary)
+            mFabColor = ContextCompat.getColor(activity, R.color.tab_rooms)
+            mFabPressedColor = ContextCompat.getColor(activity, R.color.tab_rooms_secondary)
+        }
+    }
+
+    override fun getRooms(): MutableList<Room> {
+        return ArrayList(mSession.dataHandler.store?.rooms)
+    }
+
+    override fun onFilter(pattern: String?, listener: OnFilterListener?) {
+        dataUpdateListeners.forEach {
+            it.onFilter(pattern, listener)
+        }
+    }
+
+    override fun onResetFilter() {
+        dataUpdateListeners.forEach {
+            it.onFilter("", null)
+        }
+    }
+
+    override fun onRoomResultUpdated(result: HomeRoomsViewModel.Result?) {
+        if (isResumed) {
+            refreshData(result)
+        }
+    }
+
+    /*
+     * *********************************************************************************************
+     * Data management
+     * *********************************************************************************************
+     */
+    /**
+     * Init the rooms data
+     */
+    private fun refreshData(result: HomeRoomsViewModel.Result?) {
+        this.result = result
+        val pinMissedNotifications = PreferencesManager.pinMissedNotifications(activity)
+        val pinUnreadMessages = PreferencesManager.pinUnreadMessages(activity)
+        val notificationComparator = RoomUtils.getNotifCountRoomsComparator(mSession, pinMissedNotifications, pinUnreadMessages)
+        dataUpdateListeners.forEachIndexed { index, listener ->
+            when (index) {
+                ROOM_FRAGMENTS.INVITE.ordinal -> {
+                    listener.onUpdate(mActivity.roomInvitations, notificationComparator)
+                }
+                ROOM_FRAGMENTS.FAVORITE.ordinal -> {
+                    listener.onUpdate(result?.favourites, notificationComparator)
+                }
+                ROOM_FRAGMENTS.NORMAL.ordinal -> {
+                    listener.onUpdate(result?.otherRooms?.plus(result.directChats), notificationComparator)
+                }
+                ROOM_FRAGMENTS.LOW_PRIORITY.ordinal -> {
+                    listener.onUpdate(result?.lowPriorities, notificationComparator)
+                }
+            }
```

Comment:
Please code defensively.  If there is an unknown index, please throw an exception.

Reply:
else branch has been added.